controllers/policyserver_controller_test.go

```diff
@@ -18,130 +18,261 @@ package controllers
+				}, timeout, pollInterval).ShouldNot(Succeed())
+
+				Eventually(func(g Gomega) error {
+					_, err := getTestPolicyServerService(policyServerName)
+					return err
+				}, timeout, pollInterval).ShouldNot(Succeed())
+
+				Eventually(func(g Gomega) (*policiesv1.PolicyServer, error) {
+					return getTestPolicyServer(policyServerName)
+				}, timeout, pollInterval).ShouldNot(
+					HaveField("Finalizers", ContainElement(constants.KubewardenFinalizer)),
+				)
+			})
+		})
+	})
+
+	Context("when starting policy server", func() {
+		policyServerName := newName("policy-server")
+
+		It("with MinAvailable PodDisruptionBudget configuration should create PDB", func() {
+			minAvailable := intstr.FromInt(2)
+			policyServer := policyServerFactory(policyServerName)
+			policyServer.Spec.MinAvailable = &minAvailable
 			// It's necessary remove the test finalizer to make the
-			// BeforeEach work as extected. Otherwise, the policy service
-			// creation will not work as expected
-			policyServer, err := getTestPolicyServer(policyServerName)
-			Expect(err).Should(Succeed())
+			// policy service goes away.
 			controllerutil.RemoveFinalizer(policyServer, IntegrationTestsFinalizer)
-			err = reconciler.Client.Update(ctx, policyServer)
-			Expect(err).ToNot(HaveOccurred())
+
+			Expect(
+				k8sClient.Create(ctx, policyServer),
+			).To(haveSucceededOrAlreadyExisted())
+			// Wait for the Service associated with the PolicyServer to be created
 			Eventually(func(g Gomega) error {
 				_, err := getTestPolicyServer(policyServerName)
 				return err
-			}, timeout, pollInterval).ShouldNot(Succeed())
+			}, timeout, pollInterval).Should(Succeed())
+			Eventually(func(g Gomega) error {
+				return policyServerPodDisruptionBudgetExists(policyServerName)
+			}, timeout, pollInterval).Should(Succeed())
 		})
 
-	})
-
-	Context("it has assigned policies", Serial, func() {
-		policyName := newName("policy")
+		It("with MaxUnavailable PodDisruptionBudget configuration should create PDB", func() {
+			maxUnavailable := intstr.FromInt(2)
+			policyServer := policyServerFactory(policyServerName)
+			policyServer.Spec.MaxUnavailable = &maxUnavailable
+			// It's necessary remove the test finalizer to make the
+			// policy service goes away.
+			controllerutil.RemoveFinalizer(policyServer, IntegrationTestsFinalizer)
 
-		It("should delete assigned policies", func() {
-			By("creating a policy and assigning it to the policy server")
 			Expect(
-				k8sClient.Create(ctx, clusterAdmissionPolicyFactory(policyName, policyServerName, false)),
+				k8sClient.Create(ctx, policyServer),
 			).To(haveSucceededOrAlreadyExisted())
+			// Wait for the Service associated with the PolicyServer to be created
+			Eventually(func(g Gomega) error {
+				_, err := getTestPolicyServer(policyServerName)
+				return err
+			}, timeout, pollInterval).Should(Succeed())
+			Eventually(func(g Gomega) error {
+				return policyServerPodDisruptionBudgetExists(policyServerName)
```

Comment:
We could make the tests stricter by using a `getTestPDB` utility function and testing that the values are set correctly see https://github.com/kubewarden/kubewarden-controller/blob/37073dbfcce2130626775614a1fe4a27e3f39562/controllers/admissionpolicy_controller_test.go#L91 instead of asserting the existence of the PDB.

Reply:
I've pushed a gomega matcher to address this request. It's a little bit different from what you request. During the tests I found out that if we follow the example that you sent, the tests will fail after the first failed tried in the `Eventually` call. As far as I can see, the example test is not failing only because the order of the tests. In other words, it's passing because the order of the execution and due how the controller set policy status. The webhook always exists when the test start. To fix the test, we can follow the approach that I've used in this PR. Or we should update the `Expect` calls to be `g.Expect` (using the argument of the function.

Reply:
I'll open an issue to address this problem.